src/pages/studyView/charts/ChartContainer.tsx

```diff
@@ -1228,7 +1316,50 @@ export class ChartContainer extends React.Component<IChartContainerProps, {}> {
                         />
                     );
                 };
-                break;
+            case ChartTypeEnum.MUTATION_DIAGRAM:
+                return () =>
+                    this.props.promise.isComplete ? (
+                        <div>
+                            <LollipopMutationPlot
+                                store={this.props.store.getMutationStore(
+                                    this.props.chartMeta.uniqueKey
+                                )}
+                                autoHideControls={false}
+                                showLegendToggle={false}
+                                showDownloadControls={false}
+                                showTrackSelector={false}
+                                showYMaxSlider={false}
+                                geneWidth={
+                                    this.props.store.chartsDimension.get(
+                                        this.props.chartMeta.uniqueKey
+                                    )!.w * 165
+                                }
+                                vizHeight={
+                                    this.props.store.chartsDimension.get(
+                                        this.props.chartMeta.uniqueKey
+                                    )!.h * 140
+                                }
+                                onRef={(ref: any) => {
+                                    this.mutationPlotRef = ref;
+                                }}
+                                yAxisLabelFormatter={() => {
+                                    return '';
+                                }}
+                                xAxisOnTop={
+                                    this.props.store.getMutationStore(
+                                        this.props.chartMeta.uniqueKey
+                                    ).isXAxisOnTopSelected
+                                }
+                            />
+                            {this.props.store.getMutationStore(
+                                this.props.chartMeta.uniqueKey
+                            ).samplesViaSelectedCodons.length > 0 &&
+                                this.getMutationPlotControls()}
+                        </div>
+                    ) : (
+                        <p>Loading...</p>
```

Comment:
might be better to reuse a component here, like `LoadingIndicator` or something similar